src/dispatching/repls/commands_repl.rs

```diff
@@ -12,35 +12,56 @@ use std::{fmt::Debug, marker::PhantomData};
 use teloxide_core::requests::Requester;
 
 /// A [REPL] for commands.
+//
+#[doc = include_str!("preamble.md")]
 ///
-/// All errors from an update listener and handler will be logged.
+/// [REPL]: https://en.wikipedia.org/wiki/Read-eval-print_loop
 ///
-/// REPLs are meant only for simple bots and rapid prototyping. If you need to
-/// supply dependencies or describe more complex dispatch logic, please use
-/// [`Dispatcher`].
+/// ## Signature
 ///
-/// See also: ["Dispatching or
-/// REPLs?"](dispatching/index.html#dispatching-or-repls)
+/// Don't be scared by many trait bounds in the signature, in essence they
+/// require:
 ///
-/// ## Caution
+/// 1. `bot` is a bot, client for the Telegram bot API
+///    - in teloxide this is represented via a [`Requester`] trait
+/// 2. `handler` is an async function that returns `Result<(), E>`
+///    - Such that `E` can be printed with [`Debug`] formatting
+///    - And all arguments can be extracted from [`DependencyMap`]
+///       - Which is the same, as all arguments implementing `Send + Sync +
+///         'static`
+/// 3. `cmd` is a type of the command that will be parsed,
+///    - The command type must implement [`BotCommands`] trait
+///    - It can be acquired by writing `TheCommandType::ty()`
 ///
-/// **DO NOT** use this function together with [`Dispatcher`] and other REPLs,
-/// because Telegram disallow multiple requests at the same time from the same
-/// bot.
+/// All other requirements are about thread safety and data validity and can be
+/// ignored for most of the time.
 ///
-/// ## Dependency requirements
+/// ## Handler arguments
 ///
-///  - Those of [`HandlerExt::filter_command`]
+/// Teloxide provides the following types to the `handler`:
+/// - [`Message`]
+/// - `R` (type of the `bot`)
+/// - `Cmd` (type of the parsed command)
+/// - [`Me`]
+///
+/// [`Me`]: crate::types::Me
+/// [`Message`]: crate::types::Message
+///
+/// ## Stopping
+//
+#[doc = include_str!("stopping.md")]
+///
+/// ## Caution
+//
+#[doc = include_str!("caution.md")]
 ///
-/// [REPL]: https://en.wikipedia.org/wiki/Read-eval-print_loop
-/// [`Dispatcher`]: crate::dispatching::Dispatcher
 #[cfg(feature = "ctrlc_handler")]
 pub async fn commands_repl<'a, R, Cmd, H, Args>(bot: R, handler: H, cmd: PhantomData<Cmd>)
 where
-    Cmd: BotCommands + Send + Sync + 'static,
-    H: Injectable<DependencyMap, Result<(), RequestError>, Args> + Send + Sync + 'static,
     R: Requester + Clone + Send + Sync + 'static,
     <R as Requester>::GetUpdates: Send,
+    H: Injectable<DependencyMap, Result<(), RequestError>, Args> + Send + Sync + 'static,
```

Comment:
```suggestion
    H: Injectable<DependencyMap, ResponseResult, Args> + Send + Sync + 'static,
```

Reply:
This is unrelated to this PR, let's do it in a follow-up

Reply:
Should we? It's just the same type but an alias. We've already removed some lifetime in this PR, which is not entirely related to the documentation.

Reply:
Nevermind, it'll be easier to fix docs with this